torchserve/datadog_checks/torchserve/model_discovery.py

```diff
@@ -0,0 +1,63 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.checks.kubelet_base.base import urljoin
+from datadog_checks.base.utils.discovery import Discovery
+
+
+class ModelDiscovery(Discovery):
+    def __init__(self, check, limit=None, include=None, exclude=None, interval=None):
+        super().__init__(
+            self.get_models,
+            limit=limit,
+            include={pattern: None for pattern in include} if include else None,
+            exclude=exclude,
+            interval=interval,
+            key=lambda n: n.get("modelName"),
+        )
+        self.check = check
+        self.api_status = AgentCheck.UNKNOWN
+        self.all_models = None
+        self.refreshed = False
```

Comment:
I'm not completely sure how `refreshed` works here and how it changes. Why does `get_items` turn it to `False` and `get_models` turn it to `True`?

Reply:
Added comments directly in the code, hope that help. 

It's because the `get_models` is not always called in `get_items`, it depends on the cache configuration we have

Reply:
https://github.com/DataDog/integrations-core/pull/14608/commits/8fcffb0e39bb8c1ffbe6c4fc491790ec6d40b997